Handle missing serial numbers

alexford opened the pull request:
After the discussion in #41 I figured I'd push up what I did to work around nodes without serial numbers available. I factored out that serial number retrieval into its own function.

Separately, I added a Jest test to help cover that case.

_I realize choosing and adding Jest is a rather large suggestion/change. I'm happy to revert that if you'd rather not take that step, or use a different framework, etc. It was just a lot quicker to make sure this behaves as I expected by writing tests versus poking at real Z-Wave devices._

alexford committed:
Fallback gracefully if node has no serialNumber

alexford committed:
Add jest

alexford committed:
Test for _findNodeSerialNumber

alexford committed:
return null instead of undefined from _findNodeSerialNumber

alexford committed:
standard --fix

mike182uk commented:
Hey @alexford 

Sorry for the late reply, i completely missed this 🙈

Would you mind fixing the lint error and resolving the merge conflicts? Its probably easiest to pull in the `package.json` changes from `master` and running `npm i` to regenerate the `package-lock.json` rather than trying to resolve its conflicts manually 😃

> I realize choosing and adding Jest is a rather large suggestion/change. I'm happy to revert that if you'd rather not take that step, or use a different framework, etc. It was just a lot quicker to make sure this behaves as I expected by writing tests versus poking at real Z-Wave devices.

Thanks for the taking the time to add this, its definitely something I should of got round to sooner rather than later. More than happy to keep it in 👍

alexford committed:
make linting aware of jest globals

https://stackoverflow.com/questions/46094324/javascript-standard-style-does-not-recognize-jest

alexford committed:
Add tests toCI

alexford committed:
On second thought, don't run tests in CI yet

alexford commented:
I've rebased and fixed the lint error. Tried to get the tests to run on CI but openzwave doesn't seem fully happy on Travis (https://travis-ci.org/github/mike182uk/homebridge-zwave/builds/726733400?utm_source=github_status&utm_medium=notification)

mike182uk commented:
Ah maybe we need to add a jest config file and tell it to only run tests in the `lib` dir?
